runtime/src/prioritization_fee_cache.rs

```diff
@@ -572,239 +560,232 @@ mod tests {
             .is_empty());
 
         // Assert after add one transaction for slot 1
         {
             let txs = vec![
-                build_sanitized_transaction_for_test(5, &write_account_a, &write_account_b),
+                build_sanitized_transaction_for_test(2, &write_account_a, &write_account_b),
                 build_sanitized_transaction_for_test(
-                    0,
+                    1,
                     &Pubkey::new_unique(),
                     &Pubkey::new_unique(),
                 ),
             ];
-            prioritization_fee_cache.update(bank1.clone(), txs.iter());
+            sync_update(&mut prioritization_fee_cache, bank1, txs.iter());
             // before block is marked as completed
             assert!(prioritization_fee_cache
-                .get_account_prioritization_fees(&write_account_a)
+                .get_prioritization_fees(&[])
                 .is_empty());
             assert!(prioritization_fee_cache
-                .get_account_prioritization_fees(&write_account_b)
+                .get_prioritization_fees(&[write_account_a])
                 .is_empty());
             assert!(prioritization_fee_cache
-                .get_account_prioritization_fees(&write_account_c)
+                .get_prioritization_fees(&[write_account_b])
+                .is_empty());
+            assert!(prioritization_fee_cache
+                .get_prioritization_fees(&[write_account_c])
+                .is_empty());
+            assert!(prioritization_fee_cache
+                .get_prioritization_fees(&[write_account_a, write_account_b])
+                .is_empty());
+            assert!(prioritization_fee_cache
+                .get_prioritization_fees(&[write_account_a, write_account_b, write_account_c])
                 .is_empty());
             // after block is completed
-            sync_finalize_priority_fee_for_test(&mut prioritization_fee_cache, bank1.slot());
+            sync_finalize_priority_fee_for_test(&mut prioritization_fee_cache, 1);
             assert_eq!(
-                vec![5],
-                prioritization_fee_cache.get_account_prioritization_fees(&write_account_a)
+                hashmap_of(vec![(1, 1)]),
+                prioritization_fee_cache.get_prioritization_fees(&[])
             );
             assert_eq!(
-                vec![5],
-                prioritization_fee_cache.get_account_prioritization_fees(&write_account_b)
+                hashmap_of(vec![(1, 2)]),
+                prioritization_fee_cache.get_prioritization_fees(&[write_account_a])
+            );
+            assert_eq!(
+                hashmap_of(vec![(1, 2)]),
+                prioritization_fee_cache.get_prioritization_fees(&[write_account_b])
+            );
+            assert_eq!(
+                hashmap_of(vec![(1, 1)]),
+                prioritization_fee_cache.get_prioritization_fees(&[write_account_c])
+            );
```

Comment:
Sorry to make this comment out of the blue, but I'm trying to understand the implementation of the `getRecentPrioritizationFees` JSON-RPC method (also created a [question](https://solana.stackexchange.com/questions/6091/should-getrecentprioritizationfees-always-return-0-as-fees-when-an-empty-array) in the Solana Stack Exchange page to address it).

The actual question is: at this point why should the cache contain data for `write_account_c`, since just a transaction that involves `write_account_a` and `write_account_b` has been submitted? Shouldn't it return an empty result?

Reply:
> at this point why should the cache contain data for `write_account_c`

The cache doesn't in fact contain any explicit data for that account. If a key isn't found in the cache, `get_prioritization_fees` returns the minimum fee for any transaction in that slot: https://github.com/solana-labs/solana/blob/b0112a5f43fd142ce1d7eebc39e9034aa49e9c2f/runtime/src/prioritization_fee_cache.rs#L372-L382

Reply:
@CriesofCarrots thanks for answering!

I guess that this also explain why I basically always get 0 values when not passing accounts as parameter to this method, because it means that there is always a transaction in the slot with 0 as the value for the priority fee and, since the minimum value is fetched from the cache, 0 is going to be returned. Is that explanation correct?

Reply:
Yes, correct.

Reply:
That makes more sense now :) . Don't you think that the docs for this method should specify that the values returned are minimum values? It currently mentions that the returned value for each slot is the "per-compute-unit fee paid by **at least** one successfully landed transaction, specified in increments of 0.000001 lamports".

Although this is not wrong, having "**minimum** per-compute-unit fee paid by a landed transaction" instead would sound more representative to what happens for this method, in my opinion.